scripts/conversion_script.py

```diff
@@ -222,28 +222,41 @@ def run_subprocess(cmd, print_cmd=True, env=None):
     return output, process.returncode
 
 
+def _check_if_package_installed(pkg_name):
+    _, return_code = run_subprocess(["/usr/bin/rpm", "-q", pkg_name])
+    return return_code == 0
+
+
 def install_convert2rhel():
-    """Install the convert2rhel tool to the system."""
+    """
+    Install the convert2rhel tool to the system.
+    Returns True if the yum transaction should be undone during cleanup.
+    """
     print("Installing & updating Convert2RHEL package.")
-    output, returncode = run_subprocess(
-        ["/usr/bin/yum", "install", "convert2rhel", "-y"],
-    )
-    if returncode:
-        raise ProcessError(
-            message="Failed to install convert2rhel RPM.",
-            report="Installing convert2rhel with yum exited with code '%s' and output: %s."
-            % (returncode, output.rstrip("\n")),
+
+    c2r_pkg_name = "convert2rhel"
```

Comment:
I would probably track this with a global structure, marking if a package can be uninstalled after we are done. But the way it is here will work just as fine.

Reply:
yep, just pushed changes, now using global `YUM_TRANSACTIONS_TO_UNDO = set()` with transaction ids